internal/cmd/grpc.go

```diff
@@ -105,37 +108,46 @@ func NewGRPCServer(
 		return server.ln.Close()
 	})
 
-	db, driver, err := sql.Open(*cfg)
-	if err != nil {
-		return nil, fmt.Errorf("opening db: %w", err)
-	}
+	var store storage.Store
 
-	if driver == sql.SQLite && cfg.Database.MaxOpenConn > 1 {
-		logger.Warn("ignoring config.db.max_open_conn due to driver limitation (sqlite)", zap.Int("attempted_max_conn", cfg.Database.MaxOpenConn))
-	}
+	dbStoreFn := func() (storage.Store, error) {
+		db, driver, shutdown, err := getDB(ctx, logger, cfg, forceMigrate)
+		if err != nil {
+			return nil, err
+		}
 
-	server.onShutdown(func(context.Context) error {
-		return db.Close()
-	})
+		server.onShutdown(shutdown)
+
+		switch driver {
+		case fliptsql.SQLite:
+			store = sqlite.NewStore(db, logger)
+		case fliptsql.Postgres, fliptsql.CockroachDB:
+			store = postgres.NewStore(db, logger)
+		case fliptsql.MySQL:
+			store = mysql.NewStore(db, logger)
+		default:
+			return nil, fmt.Errorf("unsupported driver: %s", driver)
+		}
 
-	if err := db.PingContext(ctx); err != nil {
-		return nil, fmt.Errorf("pinging db: %w", err)
-	}
+		logger.Debug("database driver configured", zap.Stringer("driver", driver))
 
-	var store storage.Store
+		return store, nil
+	}
 
-	switch driver {
-	case sql.SQLite:
-		store = sqlite.NewStore(db, logger)
-	case sql.Postgres, sql.CockroachDB:
-		store = postgres.NewStore(db, logger)
-	case sql.MySQL:
-		store = mysql.NewStore(db, logger)
+	switch cfg.Storage.Type {
+	case config.DatabaseStorageType:
+		store, err = dbStoreFn()
+		if err != nil {
+			return nil, err
+		}
 	default:
-		return nil, fmt.Errorf("unsupported driver: %s", driver)
+		store, err = dbStoreFn()
+		if err != nil {
+			return nil, err
```

Comment:
I would probably error here instead. We should be able to set a default to `database` and if this is something else fail.

Reply:
@GeorgeMac Yeah, understood. I thought to do it this way for full backwards compatibility, but erroring here makes sense too.